controller/Equations_Controller.java

```diff
@@ -0,0 +1,364 @@
+		myGoal = goal;
+		model.addAttribute("game", goal);
+		for (int i = 0; i < 2; i++) {
+			if (playerID == myPlayers[i])
+				myTurn = i;
+		}
+		myCurrentPlayerID = playerID;
+		passTurn();
+		return "game";
+	}
+
+	/**
+	 * This takes in view knowledge of the mat. More specifically, what the player
+	 * is requesting to be put on the 'goal' portion of the mat. Update the model
+	 * with this information. Index is the position of the die in a list of die.
+	 * 
+	 * @param index
+	 */
+	public void goalSet(int index, Die moved) {
+		// Send die to goal
+		myModel.moveDie(index, 0);
+	}
+
+	/**
+	 * Skip to the next persons turn. It should be 1 if the last turn is player 3's.
+	 */
+	public void passTurn() {
+		if (myTurn == 3) {
+			myTurn = 1;
+			myCurrentPlayerID = myPlayers[0];
+		} else {
+			myTurn++;
+			myCurrentPlayerID = myPlayers[myTurn-1];
+		}
+	}
+
+	/**
+	 * Moves a die to one of three locations, forbidden, required, or allowed.
+	 * 
+	 * @param location
+	 * @param index
+	 */
+	@GetMapping ("/game")
+	public boolean moveDie(int location, int index, Model model) {
+		switch (location) {
+		case 0:
+			Die movedF = myModel.getMyResources().getMyMat().elementAt(index);
+			this.apprise(model, movedF);
+			return this.moveToForbidden(index);
+		case 1:
+			Die movedR = myModel.getMyResources().getMyMat().elementAt(index);
+			this.apprise(model, movedR);
+			return this.moveToRequired(index);
+		case 2:
+			Die movedA = myModel.getMyResources().getMyMat().elementAt(index);
+			this.apprise(model, movedA);
+			return this.moveToAllowed(index);
+
+		}
+
+		// If we're lucky, this shouldn't occur.
+		System.out.println("ERROR");
+		return false;
+
+	}
+	
+	/**
+	 * Move die to forbidden, uses die from resources at the index position.
+	 * 
+	 * @param index
+	 * @return true, if successful.
+	 */
+	private boolean moveToForbidden(int index) {
+		myModel.moveDie(index, 1);
+		if (myModel.moveDie(index, 1) == -1) {
+			return false;
+		}
+
+		return true;
+
+	}
+
+	/**
+	 * Move die to required, uses die from resources at the index position.
+	 * 
+	 * @param index
+	 * @return true, if successful.
+	 */
+	private boolean moveToRequired(int index) {
+		myModel.moveDie(index, 2);
+		if (myModel.moveDie(index, 2) == -1) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Move die to allowed, uses die from resources at the index position.
+	 * 
+	 * @param index
+	 * @return true, if successful.
+	 */
+	private boolean moveToAllowed(int index) {
+		myModel.moveDie(index, 3);
+		if (myModel.moveDie(index, 3) == -1) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if the challenge is impossible. The function will return false if the
+	 * equation is not valid.
+	 * 
+	 * @param playerInput
+	 */
+	public boolean challengeImpossible(String playerInput) {
+		Algebra a = new Algebra(playerInput, myGoal);
+		Solver s = new Solver(a);
+		if (s.validEquation(playerInput) == false)
```

Comment:
`if (s.validEquation(playerInput) == false)`
Can be written as
`if (!s.validEquation(playerInput))`